### PR TITLE
TSK-1192 New Role for Task-Admin

### PR DESCRIPTION
**Pull request by gitgoodjhe**

### CODE REVIEW

**Comment by gitgoodjhe** [comment]
Sonarcloud: https://sonarcloud.io/dashboard?branch=TSK_1192&id=gitgoodjhe_taskana

File: lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java [hunk 1]
```diff
@@ -75,7 +75,9 @@ public Workbasket getWorkbasket(String workbasketId)
         throw new WorkbasketNotFoundException(
             workbasketId, "Workbasket with id " + workbasketId + " was not found.");
       }
-      if (!taskanaEngine.getEngine().isUserInRole(TaskanaRole.ADMIN, TaskanaRole.BUSINESS_ADMIN)) {
+      if (!taskanaEngine
```
Comment (holgerhagen) [inline lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/WorkbasketServiceImpl.java:78]: Do we need this? Since this is not directly task related I would question this part

File: lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java [hunk 2]
```diff
@@ -53,18 +55,41 @@ void should_FailToDeleteTaskComment_When_UserHasNoAuthorization()
...
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_DeleteTaskComment_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
+      throws NotAuthorizedException, TaskNotFoundException, TaskCommentNotFoundException,
+                 InvalidArgumentException, SQLException {
+
+    resetDb(false);
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java:79]: Do we have dependencies between the tests?

File: lib/taskana-core/src/test/java/acceptance/task/TransferTaskAccTest.java [hunk 3]
```diff
@@ -134,21 +158,7 @@ void testTransferTaskDoesNotExist() {
 
   @WithAccessId(user = "teamlead_1", groups = "teamlead_1")
   @Test
-  void testTransferNotAuthorizationOnWorkbasketTransfer() {
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/task/TransferTaskAccTest.java:137]: why did you delete this test?

File: lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java [hunk 4]
```diff
@@ -129,6 +130,31 @@ void should_ThrowException_When_TaskHasAlreadyBeenUpdated()
...
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getClassificationSummary().getKey()).isEqualTo("T2100");
+    assertThat(updatedTask.getClassificationSummary()).isNotEqualTo(classificationSummary);
+    assertThat(updatedTask.getCreated()).isNotEqualTo(updatedTask.getModified());
+    assertThat(task.getPlanned()).isEqualTo(updatedTask.getPlanned());
+    assertThat(task.getName()).isEqualTo(updatedTask.getName());
+    assertThat(task.getDescription()).isEqualTo(updatedTask.getDescription());
+    resetDb(false); // classification of task TKI:0..00 was changed...
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java:155]: this should not be necessary if each test works own dedicated data sets.

File: lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java [hunk 5]
```diff
@@ -129,6 +130,31 @@ void should_ThrowException_When_TaskHasAlreadyBeenUpdated()
...
+                 InvalidStateException, SQLException {
+
+    TaskService taskService = taskanaEngine.getTaskService();
+    Task task = taskService.getTask("TKI:000000000000000000000000000000000000");
+    final ClassificationSummary classificationSummary = task.getClassificationSummary();
+    task.setClassificationKey("T2100");
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java:148]: in this testcase we do not need all the assertions, do we? I assume we have this tested in another testcase?

File: lib/taskana-core/src/test/java/acceptance/workbasket/DeleteWorkbasketAuthorizationsAccTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          workbasketService.deleteWorkbasketAccessItemsForAccessId("group_1");
+        };
+
+    assertThatThrownBy(deleteWorkbasketAccessItemCall).isInstanceOf(NotAuthorizedException.class);
+
+    WorkbasketAccessItem workbasketAccessItem =
+        workbasketService.newWorkbasketAccessItem(
+            "WBI:100000000000000000000000000000000008", "newAccessIdForUpdate");
+
+    workbasketAccessItem.setPermCustom1(true);
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/workbasket/DeleteWorkbasketAuthorizationsAccTest.java:40]: what do we need this for?

File: lib/taskana-core/src/test/java/acceptance/workbasket/GetWorkbasketAccTest.java [hunk 7]
```diff
@@ -51,6 +52,38 @@ void testGetWorkbasketById() throws NotAuthorizedException, WorkbasketNotFoundEx
...
+
+    assertThat(retrievedWorkbasket).isNotNull();
+    assertThat(retrievedWorkbasket.getOwner()).isEqualTo("Peter Maier");
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "businessadmin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_ReturnWorkbasketById_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/workbasket/GetWorkbasketAccTest.java:76]: see above. I'm not sure if we want to do this.

File: lib/taskana-core/src/test/java/acceptance/workbasket/GetWorkbasketAuthorizationsAccTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Acceptance test for all "get workbasket authorizations" scenarios.
+ */
+@ExtendWith(JaasExtension.class)
+public class GetWorkbasketAuthorizationsAccTest extends AbstractAccTest {
+
+  @WithAccessId(user = "user_1_1")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  public void should_ThrowException_When_UserRoleIsNotAdminOrBusinessAdmin() {
```
Comment (holgerhagen) [inline lib/taskana-core/src/test/java/acceptance/workbasket/GetWorkbasketAuthorizationsAccTest.java:24]: again - but here we already reject the get? Doesn't look consistent to me. But I would rather change the other parts.

File: lib/taskana-core/src/test/java/acceptance/workbasket/DeleteWorkbasketAuthorizationsAccTest.java (hunk 6 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/workbasket/DeleteWorkbasketAuthorizationsAccTest.java:40]: changed it

File: lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java (hunk 5 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java:148]: changed it

File: lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java (hunk 4 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/task/UpdateTaskAccTest.java:155]: Was able to remove this without repurcussions - looks like it was indeed unnecessary

File: lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java (hunk 2 above)
Comment (gitgoodjhe) [reply lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java:79]: This test runs twice. It deletes a task comment which can't be deleted twice, so a reset is necessary.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java [hunk 9]
```diff
@@ -70,7 +70,8 @@ TaskComment updateTaskComment(TaskComment taskCommentToUpdate)
       taskService.getTask(taskCommentImplToUpdate.getTaskId());
 
       if (taskCommentToUpdate.getCreator().equals(userId)
```
[the remark this answers is not in the document]
Comment (gitgoodjhe) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java:72]: Indeed. Will fix this in a separate ticket though

File: lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java (hunk 2 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java:79]: > Each invocation of a test template method behaves like the execution of a regular @Test method with full support for the same lifecycle callbacks and extensions.

https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-templates

Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/DeleteTaskCommentAccTest.java:79]: I see the issue: resetDb(..) is executed once for each TestClass.
Idea:
Depending on the current user try to delete a different ID and provide two Ids within the test-data
